plugins/catalog-graph/src/components/CatalogGraphPage/useCatalogGraphPage.test.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { RELATION_MEMBER_OF } from '@backstage/catalog-model';
+import { act, renderHook } from '@testing-library/react-hooks';
+import { useLocation as useLocationMocked } from 'react-router';
+import { Direction } from '../EntityRelationsGraph';
+import { useCatalogGraphPage } from './useCatalogGraphPage';
+
+jest.mock('react-router', () => ({
+  useLocation: jest.fn(),
+}));
+
+jest.spyOn(window.history, 'replaceState');
+jest.spyOn(window.history, 'pushState');
+
+const useLocation = useLocationMocked as jest.Mock;
```

Comment:
is it possible to make these `jest.Mock<typeof useLocationMocked>` to get some typeness? i like the pattern as such but maybe this would be even neater

Reply:
A bit more challenging, but `jest.Mock<ReturnType<typeof useLocationMocked>>` works.